coriolis_openstack_utils/resource_utils/routers.py

```diff
@@ -0,0 +1,143 @@
+from oslo_log import log as logging
+
+from coriolis_openstack_utils import conf
+from coriolis_openstack_utils.resource_utils import subnets
+from coriolis_openstack_utils.resource_utils import networks
+
+CONF = conf.CONF
+LOG = logging.getLogger(__name__)
+
+
+def get_router(openstack_client, name_or_id):
+    return openstack_client.neutron.find_resource('router', name_or_id)
+
+
+def list_routers(openstack_client, filters):
+    return openstack_client.neutron.list_routers(**filters)['routers']
+
+
+def check_router_similarity(source_client, src_router, destination_client,
+                            dest_router):
+    relevant_keys = {'admin_state_up', 'external_gateway_info',
+                     'distributed', 'ha'}
+    conflicting_keys = set()
+
+    router_network_mapping = CONF.destination.external_network_name_map
+    src_router = {k: v for k, v in src_router.items() if v is not None}
+    dest_router = {k: v for k, v in dest_router.items() if v is not None}
+    for k in src_router:
+        if k in relevant_keys:
+            if k == 'external_gateway_info':
+                src_snat = src_router[k].get('enable_snat', False)
+                dest_snat = dest_router.get(k, {}).get('enable_snat', False)
+                src_network_id = src_router[k]['network_id']
+                dest_network_id = dest_router.get(
+                    k, {}).get('network_id', True)
+                if src_snat == dest_snat:
+                    src_net_name = networks.get_network(
+                        source_client, src_network_id)['name']
+                    dest_net_name = networks.get_network(
+                        destination_client, dest_network_id)['name']
+                    if (router_network_mapping.get(src_net_name, src_net_name)
+                            == dest_net_name):
+                        conflicting_keys.add(k)
+            elif src_router[k] == dest_router.get(k):
+                conflicting_keys.add(k)
+
+    src_relevant_keys = set(src_router.keys()).intersection(relevant_keys)
+    return conflicting_keys == src_relevant_keys
+
+
+def get_migration_info(source_client, name_or_id):
+    router = get_router(source_client, name_or_id)
+    relevant_keys = {'admin_state_up', 'distributed', 'ha',
+                     'availability_zone_hints'}
+
+    body = {k: v for k, v in router.items() if k in relevant_keys}
+    body['availability_zone_hints'] += router['availability_zones']
+    # eliminate any possible duplicates
+    body['availability_zone_hints'] = list(
+        set(body['availability_zone_hints']))
+    ext_gateway_info = router['external_gateway_info']
+
+    src_ext_subnet_ids = set()
+    for fixed_ip in ext_gateway_info['external_fixed_ips']:
+        src_ext_subnet_ids.add(fixed_ip['subnet_id'])
+    # mapped with external_network_name_map
```

Comment:
As you're not actually doing the mapping in this function, please either make this comment more explicit (ex: "These subnet names **will** get mapped bla bla...", or remove the comment entirely.